src/pool/srv_pool.c

```diff
@@ -2862,6 +2896,17 @@ ds_pool_query_handler(crt_rpc_t *rpc)
 						entry->dpe_val,
 						iv_entry->dpe_val);
 					rc = -DER_IO;
+					}
+				break;
+			case DAOS_PROP_PO_POLICY:
+				D_ASSERT(strlen(entry->dpe_str) <=
+					 DAOS_PROP_POLICYSTR_MAX_LEN);
```

Comment:
And again here.

As a reminder D_ASSERT() should be used for checking developer assumptions or implementation correctness, it should not be used for input validation.  What's the rationale for having D_ASSERT() here, and if there's any demonstrable way that it could fail which isn't the result of a bug then you should be returning an error instead.

Reply:
D_ASSERT is used here solely to catch a bug or corruption. There is no way it would fail otherwise here.